src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -343,4 +349,10 @@ private void uploadCoreDeviceCAs(List<String> certificatePemList) {
                     String.format("Failed to put core %s CA certificates to cloud", thingName), e);
         }
     }
+
+    private static List<Class> uploadCARetryExceptions() {
+        return Arrays.asList(ThrottlingException.class,
+                InternalServerException.class,
+                AccessDeniedException.class);
```

Comment:
Careful about retrying access denied. Double check what we're logging for failure. I don't want to emit a log every second when the permissions aren't changing.

Reply:
At info level, we are logging
`put-core-ca-certificate. task failed and will be retried.` only once for the first attempt. But at debug level, we log forevery attempt.